integration/maptest/map_test.go

```diff
@@ -0,0 +1,383 @@
+// Copyright 2017 Google Inc. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package maptest
+
+import (
+	"bytes"
+	"context"
+	"crypto"
+	"encoding/hex"
+	"fmt"
+	"testing"
+
+	"github.com/google/trillian"
+	"github.com/google/trillian/crypto/keys"
+	"github.com/google/trillian/merkle"
+	"github.com/google/trillian/merkle/hashers"
+	"github.com/google/trillian/testonly"
+	"github.com/google/trillian/testonly/integration"
+
+	tcrypto "github.com/google/trillian/crypto"
+	stestonly "github.com/google/trillian/storage/testonly"
+
+	_ "github.com/google/trillian/merkle/coniks"
+	_ "github.com/google/trillian/merkle/maphasher"
+)
+
+// createHashKV returns a []*trillian.MapLeaf formed by the mapping of index, value ...
+// createHashKV panics if len(iv) is odd. Duplicate i/v pairs get over written.
+func createMapLeaves(iv ...[]byte) []*trillian.MapLeaf {
```

Comment:
I was referring to `createBatchLeaves`.  I think it's a nice helper method for creating clean table based tests.  It saves me from typing `trillian.MapLeaf` over and over... It seemed useful since [gRPC metadata](https://godoc.org/google.golang.org/grpc/metadata#Pairs) did something very similar. If you feel strongly, I could expand the table tests with `trillian.MapLeaf` directly.